### DISCUSSION TITLE
Pan v3 with unlocked soc for sale

### DISCUSSION THREAD

**Discussion by gtxaspec** [gtxaspec/wz_mini_hacks#764] [General]
https://ronshed.com/product/t31x-wyze-cam-pan-v3-unlocked-soc-with-thingino-firmware/

My buddy Ron does component level electronics work, and has successfully transplanted a non secure boot ingenic CPU to the pan v3.  

This unit doesn't run wyze software, it has the [thingino](https://github.com/themactep/thingino-firmware) custom firmware.  

He's only got one right now, but we've confirmed it works well!

Just FYI for anyone interested.  The price is fair for the level of work required.

**Comment by endertable** [comment]
This is very interesting, the question is, can Wyze firmware be burned back onto  the camera.  I imagine SecureBoot and the Wyze FW are unrelated after the camera gets past the initial boot process ?!?

**Reply by gtxaspec** [reply]
yes, you should be able to put the original firmware back on if desired.  you would need a complete backup of the stock chip.

**Reply by CapnRon** [reply]
Yes, you can reflash to stock. I can put stock FW back if it is preferred, just let me know when placing the order and I'll reflash it before shipping.

**Reply by endertable** [reply]
Hi @CapnRon ,

I just wanted to clarify, I wouldn’t  want it back in its original locked state, I was more inferring to be able to put the Wyze firmware back on there, but having a telnet or SSH access to it.

thanks.

**Reply by CapnRon** [reply]
The firmware doesn't lock the SOC, that requires applying a voltage to a specific pin to damage the e-fuses. The SOC is a brand new T31X that does not have damaged e=fuses so it is capable of running the factory firmware with, or without wz mini, Thingino or any other Ingenic t31 firmware. 

It is like you said "SecureBoot and the Wyze FW are unrelated after the camera gets past the initial boot process". The new SOC does not have damaged e-fuses so it accepts any compatible firmware, where the factory SOC with the damaged e-fuses will only accept the factory wyze firmware.

**Reply by vasquezmi** [reply]
Let me make sure I am reading this correctly. 
Wyze purposely applied a specific voltage to a pin on the SOC of their product and then packaged for sale to bypass the ability to right to repair. Like for CA law.

**Reply by gtxaspec** [reply]
Many vendors do this.  They burn a specific encryption key to the device, or in this case, the processor, that can only be written once, so it will only boot their firmware, that they distribute, signed with their matching encryption key.  It's a one way process that is irreversible.

I'm not sure what the specifics are in relation to the RTR laws in CA, but that's how the newer wyze devices work.

**Reply by CapnRon** [reply]
I don't think they are specifically trying to subvert the RTR law, but it is just a beneficial side effect for them. I think their main goal is locking the device to only allow the use of their paid cloud based service. I'm sure if they are called out on it they will claim it is for  "safety and security".

**Comment by CapnRon** [comment]
![image](https://github.com/gtxaspec/wz_mini_hacks/assets/109708692/68231668-18ce-49b4-bebd-a4bb80a86ae8)

**Comment by occultrefuse** [comment]
Hello @CapnRon,

Are these still available? The link above isn't working.

**Comment by JoshuaSeidel** [comment]
> https://ronshed.com/product/t31x-wyze-cam-pan-v3-unlocked-soc-with-thingino-firmware/

The website doesnt work @gtxaspec says error with database connection. are these still available for sale?

**Comment by phantum29** [comment]
Since url is still down, has any one at least found a source for the non-secure T31x? I can't seem to find just the soc. I might try to grab a board with one from ali and transplant it too, but I'd rather not waste all the other components and subject the soc to at least two heat cycles.

**Reply by gtxaspec** [reply]
I believe ron is on sabbatical for a while.

https://www.aliexpress.us/item/3256805512975338.html
https://www.aliexpress.us/item/3256806970078742.html
T31ZX or X is preferred.

**Reply by endertable** [reply]
@phantum29 curious to after you change out the SOC, how will you get in to update the firmware or access the OS? Does this still use the wz_mini option of a kernel on the SD card?

**Reply by gtxaspec** [reply]
The bootloader/OS doesn't care about the SOC ( as long as it matches, Z/ZX )
Theres no validation from the SOC any longer.

**Comment by endertable** [comment]
@phantum29 curious to see how your venture goes with changing out the Soc

**Reply by phantum29** [reply]
I'll definitely post an update. I'm waiting on some T31x's to arrive.

**Reply by phantum29** [reply]
![000005](https://github.com/user-attachments/assets/40ab3d4f-a558-4762-84e7-1f33bc07d53b)
![000013](https://github.com/user-attachments/assets/ed385af0-6a18-464a-9bd0-e39e40f2c000)
Replaced the SOC, and had to use the ingenic USB cloner tool. The second photo is of the flash chip, and the pins match the chip referenced in the guide. Fortunately, someone made the wyze cp3 firmware and that worked out of the box, so to speak, for me.

**Reply by gtxaspec** [reply]
cool! a few handy folks like yourself have done the swap on discord already too, feel free to join in if you like in the conversation.  Hope to get some chips and rework equipment to do this myself soon.

**Reply by s00500** [reply]
nice stuff, what guide do you refer to ? is there a need to change the flash too ? I just ordered some SOCs, wanna try the replace too

**Comment by endertable** [comment]
@phantum29 that’s cool, so what equipment do you use to be able to take all those little solders off the legs? Also, you were saying you found some Wyze CP3 software to use with cloner tool, can you disclose that location. I am looking at also trying this out depending on the difficulty. I am pretty handy, but also know my limits, lol.   Just curious if this project is within those limits

**Reply by gtxaspec** [reply]
The thingino firmware for the cp3 is available as a precompiled binary ready to flash in the repo.

**Comment by endertable** [comment]
@phantum29 How has the cam worked so far.  Are you using Wyze system or did you go cloudless?  Anything far-fetched in your observations?

**Reply by phantum29** [reply]
It's working great! I've done a few more, and one of them I must have messed up some how as I have no y axis control. Using it with Frigate, and it complains that the fw doesn't support `getstatus` but if I just delete the motor calibration from frigate after a reboot and reboot the camera too it works.

**Comment by CapnRon** [comment]
I’m in Philippines now, but I just ordered some new gear. If there is
enough interest I may start making small batches of these again. Let
me know what you think

**Reply by shalakolee** [reply]
I am interested

**Reply by Cook13s7777** [reply]
I would be interested as well. I have one already but perhaps a trade and pay for labor/shipping?

**Comment by s00500** [comment]
I managed to get this to work! took a few tries, but eventually did... will do a second attempt on a new camera in a few weeks I think, maybe video document that one

**Reply by gtxaspec** [reply]
glad to hear, congrats!

**Reply by endertable** [reply]
@s00500 I would be interested in that video doc.  :)   And if not the video doc, maybe some pointers.  Did you place Wyze binaries on there or did you go with thingino.  I am way invested in Wyze ATM with their  remote APP view and their AI works pretty good for me as well, but I also like to supplement with my own stuff that works with theirs as well.

**Comment by CapnRon** [comment]
The original Wyze FW will still work on the chip swapped camera.

**Comment by cmgenius** [comment]
I'm super interested in trying this. Any other known sources for getting the proper SOC in the US? Not sure if I can do an AliExpress order with the looming import rule changes ;/

**Comment by s00500** [comment]
Ok, I finally got around of making a video! so here is how this surgery worked for me:

https://youtu.be/YtA7Fi1IllI

**Comment by CapnRon** [comment]
Awesome work and nice video, its good to see others doing this hack now!
